pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2123,11 +2124,11 @@ protected CompletableFuture<Void> addReplicationCluster(String remoteCluster, Ma
             String localCluster) {
         return AbstractReplicator.validatePartitionedTopicAsync(PersistentTopic.this.getName(), brokerService)
                 .thenCompose(__ -> brokerService.pulsar().getPulsarResources().getClusterResources()
-                        .getClusterAsync(remoteCluster)
-                        .thenApply(clusterData ->
-                                brokerService.getReplicationClient(remoteCluster, clusterData)))
-                .thenAccept(replicationClient -> {
-                    if (replicationClient == null) {
+                        .getClusterAsync(remoteCluster))
+                .thenAccept((clusterData) -> {
+                    PulsarClient replicationClient = brokerService.getReplicationClient(remoteCluster, clusterData);
```

Comment:
Maybe checking partitions in the class `org.apache.pulsar.broker.admin.v2.PersistentTopics` is better

Reply:
#24118 has been implemented to check partitions. 

When the cluster version is different, this check is terrific.